opm/simulators/utils/DamarisOutputModule.hpp

```diff
@@ -34,7 +47,13 @@ namespace Opm::DamarisOutput
 {
  // Initialize an XML file
  std::string initDamarisXmlFile();
- // Initialize Damaris by filling in th XML file and stroring it in the chosed directory
- void initializeDamaris(MPI_Comm comm, int mpiRank, std::string OutputDir, bool enableDamarisOutputCollective);
+ 
+/**
+*   Initialize Damaris by either:
+*  1/ Filling in a templated XML file and storing it in the chosen directory (output directory)
+*  2/ Reading a file specified by the environment variable FLOW_DAMARIS_XML_FILE 
+*  
+*/
+ void initializeDamaris(MPI_Comm comm, int mpiRank, std::map<std::string, std::string>& find_replace_map );
```

Comment:
I tried and get a obtuse C++ error when I try to access a map entry:
```bash
 error: passing ‘const std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >’ as ‘this’ argument discards qualifiers [-fpermissive]
   68 |         std::string outputDir = find_replace_map["_PATH_REGEX_"];
```

Reply:
I can explain that one: you cannot use operator[] to access a map since that is not a const function! This is because it can cause a new element to appear. Even on the rhs like this:
```
std::map<int,int> m;
int x = m[3];
```
After this, there is one element in the map, with key 3 and value undefined or 0 (not sure exactly).

So in a const context you should use m.at("KEY_STRING") instead. This will never insert anything, and throw out_of_range if the key was not present in the map.

Reply:
> you cannot use `operator[]` to access a map since that is not a `const` function! This is because it can cause a new element to appear.

If we're being **extremely** pedantic, then `map<>::operator[]()` will **always** (attempt to) insert a new node/element into the `map`.  In particular, the subscript operator will always **create** (i.e., allocate and construct) a new element, but it will drop that new element on the floor if another element with a matching key already exists in the map in which case it will return a reference to that other element.  Otherwise, `operator[]()` will link that new element into its internal structures and return a reference to it.

> Edit: the above with `at()`:
> 
> ```c++
> std::map<int,int> m;
> int x = m.at(3); // throws as it should!
> ```

Do note, however, that the diagnostic message generated by a failing `.at()` call will often refer to internal implementation details of the map&ndash;e.g., libstdc++'s message will often refer to `_Map_base` or something like that&ndash;which will be very unfriendly if it makes its way to a user.  `map<>::at()` is usually only appropriate if the map is fully constructed/controlled by the application and does not have any user-controlled data in it.

Reply:
map.at("KEY") fixed the access issue.